### ISSUE TITLE
Compiler panic: "z must be a number literal" when sharing a state property with animated properties

### ISSUE THREAD

**Issue by RyOkEeeesh** [slint-ui/slint#11622]
### Bug Description

**What is the bug?**
The Slint compiler fails with a "must be a number literal" error or a panic during the build process when the `z` property is controlled via a conditional expression or `states`, but ONLY if another property in the same component is being animated.

**Expected behavior:**
The `z` property should be able to switch between literal values (e.g., `1` and `100`) when a state changes, as long as it is not being interpolated itself.

**Actual behavior:**
The compiler throws `error: 'z' must be an number literal` even when the values are clearly literals. This error is triggered by the presence of an `animate` block for a different property (like `width`).

**Error message:**
`Slint build failed: CompileError(["app.slint:271: 'z' must be an number literal"])
process didn't exit successfully (exit status: 101)`

### Reproducible Code (if applicable)

```slint
export component App inherits Rectangle {
    in-out property <bool> is_open: true;

    // This causes: error: 'z' must be an number literal
    z: is_open ? 100 : 1; 

    width: is_open ? 200px : 100px;
    
    // If you comment out this animation, the error for 'z' disappears.
    animate width { duration: 250ms; } 
}
```

### Environment Details

- Slint Version: 1.16.1 
- OS: macOS 26.4.1
- Rust Version: 1.94.0
- Backend/renderer: FemtoVG / Skia (Default)

### Product Impact

**What you’re building**: I'm building a dashboard UI for an embedded system.
**How this affects you**: It's a blocker for implementing app-opening animations with proper z-layering.

**Comment by ogoffart** [comment]
Thanks for opening a bug.

dynamic z ordering is not yet supported.
This is tracked in https://github.com/slint-ui/slint/issues/221 , feel free to add your 👍  there.
